### PR TITLE
refactor: Implemented using the standard library

### PR DESCRIPTION
**Pull request by aimuz**
Signed-off-by: aimuz <mr.imuz@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

/kind cleanup

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

This pr may not be of much value, but I think it is a better practice to use the standard library implementation

### CODE REVIEW

**Comment by zmerlynn** [comment]
@aimuz Sadly retest doesn't work in this repo but I kicked off a new test.

This LGTM. We're in feature freeze and this is more of a cleanup, so I'm holding off until thaw to merge.

**Comment by zmerlynn** [comment]
Bleh. Both of these failed on:

```
/usr/share/dotnet/sdk/3.1.425/NuGet.targets(128,5): error : '1.28.0-0361074' is not a valid version string. (Parameter 'value') [/go/src/agones.dev/agones/sdks/csharp/test/csharp-sdk-test.csproj]
```

which is #2018.

**Comment by zmerlynn** [comment]
@aimuz You'll need to force-push this branch again to get the commit hash to change to get past that flake. It looks like you need to rebase anyways, so you could do that, and it'll retest after you push the rebase. If it again fails in the same place, you can re-push after `git commit --amend --date=now --no-edit` to just bump the time.

**Comment by markmandel** [comment]
We can also wait until after stable, and update the branch when something else is merged. Should be fine.

File: pkg/util/signals/signals.go
Comment (roberthbailey) [inline pkg/util/signals/signals.go:29]: @markmandel - I noticed that the library function creates a channel with size 1 regardless of how many signals are being tracked: https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/os/signal/signal.go;l=284. I assume that since it's part of the std library and the API is designed to handle multiple signals that it is safe (or even correct) to set the channel to size 1 but wanted to know if there was a reason that 5 years ago you set this to size 2: https://github.com/googleforgames/agones/commit/726af15d4d8b3d40435b74ea9bfa41d1607e2881#diff-e7b27b1d977569544a1feedd454f855f61c641a2f35bf9fe75d4bc15c7cefb35

Comment (markmandel) [reply pkg/util/signals/signals.go:29]: Honestly - not that I can remember! $10 says I copied from some other example that copied from another example?

I'd give more confidence to the Go std library than code I wrote 5 years ago 😄

**Review by zmerlynn — APPROVED** [review]

**Pull request merged**
